### DISCUSSION TITLE
How to customize progressive enhancement?

### DISCUSSION THREAD

**Discussion by backstromjoel** [huntabyte/shadcn-svelte#360] [Help]
I'm trying to somehow intercept the FormAction to add some more data to the request (I need to send both the form data and some extra stuff to the server). 

After doing a bit of research, [customizing use:enhance ](https://kit.svelte.dev/docs/form-actions#progressive-enhancement-customising-use-enhance) seems to be a way to do this. How do i configure my Form.Roots use:enhance or are there other approaches to intercept and add data to a form action?

**Comment by backstromjoel** [comment]
After doing some research i found that Form.Root has a prop called options which exposes a few different callbacks onSubmit, onResult etc. which allow you to intercept and manipulate the data as you wish.

**Reply by ytsruh** [reply]
This was super helpful and I think it should be added to the documentation. I know the reference to superforms is refernce but being explicit would be super helpful
